libs/wingsdk/src/target-tf-aws/queue.ts

```diff
@@ -118,6 +120,20 @@ export class Queue extends cloud.QueueBase {
   private envName(): string {
     return `QUEUE_URL_${this.node.addr.slice(-8)}`;
   }
+
+  /**
+   * Queue names are limited to 80 characters.
+   * You can use alphanumeric characters, hyphens (-), and underscores (_).
+   */
+  private sanitizeName(name: string): string {
+    const nameProps: NameOptions = {
```

Comment:
nit: I would just put this as a const outside the class and use `ResourceNames.of()` directly in the ctor:

```ts
const NAME_OPTS = {
  maxLen: 80,
  regexMatch: /.../,
  charReplacer: "-"
};

export class Queue ... {
  constructor(...) {
    // ...
    this.queue = new SqsQueue(this, "Default", {
      name: ResourceNames.of(this, NAME_OPTS),
      // ----------------------^
      // see comment below about passing the construct so that the logic
      // that combines the id and addr is centralized under ResourceNames.
    });
  }
}
```